src/lib.rs

```diff
@@ -90,12 +166,18 @@ where
     }
 }
 
-#[derive(Debug, PartialEq)]
+/// A `ConnectionStatus` describes the current sub-status of a `Connected` connection.
+#[derive(Debug, PartialEq, Clone, Copy)]
```

Comment:
`Copy` implementations generally allow the compiler to make more aggressive optimizations, and are useful on pretty much any type that you don't implement `Drop` on.